packages/components/src/components/single-select/shadow.tsx

```diff
@@ -0,0 +1,658 @@
+		}
+		this._isOpen = false;
+	}
+
+	private clearSelection() {
+		if (this.state._disabled) {
+			return;
+		} else {
+			this._focusedOptionIndex = -1;
+			this.state._value = '';
+			this._value = '';
+			this._inputValue = '';
+			this._filteredOptions = [...this.state._options];
+
+			this.controller.setFormAssociatedValue(this._value);
+		}
+	}
+
+	private selectOption(event: Event, option: Option<string>) {
+		this.state._value = option.value;
+		this._value = option.value;
+		this._inputValue = option.label as string;
+		this.controller.onFacade.onChange(event, option.value);
+		this.controller.onFacade.onInput(event, false, option.value);
+
+		this._filteredOptions = [...this.state._options];
+
+		this.controller.setFormAssociatedValue(this._value);
+	}
+
+	private onInput(event: Event) {
+		const target = event.target as HTMLInputElement;
+		this._inputValue = target.value;
+		this._isOpen = true;
+		this.setFilteredOptionsByQuery(target.value);
+		this._focusedOptionIndex = -1;
+	}
+
+	private handleKeyDownDropdown(event: KeyboardEvent) {
+		if (event.key.length === 1 && /[a-z0-9]/i.test(event.key)) {
+			event.preventDefault();
+			this._isOpen = true;
+			this.focusSuggestionStartingWith(event.key);
+		}
+	}
+
+	private setFilteredOptionsByQuery(query: string) {
+		if (query?.trim() === '') {
+			this._filteredOptions = [...this.state._options];
+		} else if (Array.isArray(this.state._options) && this.state._options.length > 0 && query.length > 0) {
+			this._filteredOptions = this.state._options.filter((option) => {
+				return (option.label as string)?.toLowerCase()?.includes(query?.toLowerCase());
+			});
+		}
+	}
+
+	private _focusedOptionIndex: number = -1;
+
+	private moveFocus(delta: number) {
+		if (!this._filteredOptions) {
+			return;
+		}
+		let newIndex = this._focusedOptionIndex + delta;
+
+		if (newIndex >= this._filteredOptions.length) {
+			newIndex = 0;
+		}
+
+		if (newIndex < 0) {
+			newIndex = this._filteredOptions.length - 1;
+		}
+
+		this._focusedOptionIndex = newIndex;
+		this.focusOption(this._focusedOptionIndex);
+	}
+
+	private focusOption(index: number) {
+		if (this.refOptions) {
+			const optionElement = this.refOptions[index];
+			optionElement?.focus();
+		}
+	}
+
+	private focusSuggestionStartingWith(char: string) {
+		const charLowerCase = char.toLowerCase();
+
+		const index =
+			Array.isArray(this._filteredOptions) && this._filteredOptions.findIndex((option) => (option.label as string).toLowerCase().startsWith(charLowerCase));
+
+		if (typeof index === 'number') {
+			this._focusedOptionIndex = index;
+			this.focusOption(index);
+		}
+	}
+
+	public render(): JSX.Element {
+		const hasExpertSlot = showExpertSlot(this.state._label);
+		const { ariaDescribedBy } = getRenderStates(this.state);
+
+		return (
+			<Host class="kol-single-select">
+				<div class={`single-select ${this.state._disabled === true ? 'disabled' : ''} `} tabIndex={this.state._disabled ? -1 : 0}>
```

Comment:
Ich fürchte, den tabindex können wir hier nicht setzen, weil das `div` selber nicht fokussierbar ist.
Das führt auch dazu, dass ich in der React Sample App nicht mit Tab direkt zwischen den ersten zwei Beispiel-SingleSelect wechseln kann. Ich muss einmal zu viel Tab drücken und zwischendurch ist kein fokussiertes Element sichtbar.